python/ray/scripts/scripts.py

```diff
@@ -1414,6 +1427,235 @@ def status(address, redis_password):
+    is_flag=True,
+    default=True,
+    help="Increase process information verbosity")
+def get_local_logs(stream: bool = False,
+                   output: Optional[str] = None,
+                   logs: bool = True,
+                   pip: bool = True,
+                   processes: bool = True,
+                   processes_verbose: bool = False):
+    """Collect local data and package into an archive.
+
+    Usage:
+
+        ray get-local-logs [--stream/--output file]
+
+    This script is called on remote nodes to fetch their data.
+    """
+    if not get_logs_available:
+        logger.error("This function is only available in python >= 3.7")
+        return
+
+    if stream and output:
+        raise ValueError(
+            "You can only use either `--output` or `--stream`, but not both.")
+
+    parameters = GetParameters(
+        logs=logs,
+        pip=pip,
+        processes=processes,
+        processes_verbose=processes_verbose,
+        processes_list=RAY_PROCESSES)
+
+    with Archive() as archive:
+        get_all_local_data(archive, parameters)
+
+    tmp = archive.file
+
+    if stream:
+        with open(tmp, "rb") as fp:
+            os.write(1, fp.read())
+        os.remove(tmp)
+        return
+
+    target = output or os.path.join(os.getcwd(), os.path.basename(tmp))
+    os.rename(tmp, target)
+    logger.info(f"Created local data archive at {target}")
+
+
+@cli.command()
+@click.argument("cluster_config_file", required=False, type=str)
+@click.option(
+    "--host",
+    "-h",
+    required=False,
+    type=str,
+    help="Single or list of hosts, separated by comma.")
+@click.option(
+    "--ssh-user",
+    "-U",
+    required=False,
+    type=str,
+    default=None,
+    help="Username of the SSH user.")
+@click.option(
+    "--ssh-key",
+    "-K",
+    required=False,
+    type=str,
+    default=None,
+    help="Path to the SSH key file.")
+@click.option(
+    "--docker",
+    "-d",
+    required=False,
+    type=str,
+    default=None,
+    help="Name of the docker container, if applicable.")
+@click.option(
+    "--local",
+    "-L",
+    required=False,
+    type=bool,
+    is_flag=True,
+    default=None,
+    help="Also include information about the local node.")
+@click.option(
+    "--output",
+    "-o",
+    required=False,
+    type=str,
+    default=None,
+    help="Output file.")
+@click.option(
+    "--logs/--no-logs",
+    is_flag=True,
+    default=True,
+    help="Collect logs from ray session dir")
+@click.option(
+    "--pip/--no-pip",
+    is_flag=True,
+    default=True,
+    help="Collect installed pip packages")
+@click.option(
+    "--processes/--no-processes",
+    is_flag=True,
+    default=True,
+    help="Collect info on running processes")
+@click.option(
+    "--processes-verbose/--no-processes-verbose",
+    is_flag=True,
+    default=True,
+    help="Increase process information verbosity")
+def get_logs(cluster_config_file: Optional[str] = None,
```

Comment:
Can we call it `core-dump`?

Reply:
Hmm I am seeing our core dump to be more internal information (e.g., ray memory output, some debug strings, and etc.) rather than dump of logs (though I don't object your idea to call it core dump and include logs eventually).

Reply:
isn't the point/use-case of this supposed to be for getting internal information? (I think we should add everything you mentioned to this command :p )